From: SeongJae Park <sj@kernel.org>
Date: Wed, 19 Nov 2025 21:30:40 -0800
Subject: [PATCH] selftests/damon/wss_estimation: increase allowed error rate

The test's max error rate was set only with gut feeling on the test
machine that I was using before.  On a smaller machine I use nowadays,
the test fails since the working set is found slower.  Increase the
allowed error rate by increasing the percentile to evaluate from p50 to
p75.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 ...te_schemes_tried_regions_wss_estimation.py | 22 +++++++++----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/tools/testing/selftests/damon/sysfs_update_schemes_tried_regions_wss_estimation.py b/tools/testing/selftests/damon/sysfs_update_schemes_tried_regions_wss_estimation.py
index 90ad7409a7a6..a000982c59be 100755
--- a/tools/testing/selftests/damon/sysfs_update_schemes_tried_regions_wss_estimation.py
+++ b/tools/testing/selftests/damon/sysfs_update_schemes_tried_regions_wss_estimation.py
@@ -38,18 +38,18 @@ def main():
                 kdamonds.kdamonds[0].contexts[0].schemes[0].tried_bytes)
 
     wss_collected.sort()
+    percentile = 75
+    sample = wss_collected[int(len(wss_collected) * percentile / 100)]
+    error_rate = abs(sample - sz_region) / sz_region
+    print('%d-th percentile (%d) error %f' %
+            (percentile, sample, error_rate))
     acceptable_error_rate = 0.2
-    for percentile in [50, 75]:
-        sample = wss_collected[int(len(wss_collected) * percentile / 100)]
-        error_rate = abs(sample - sz_region) / sz_region
-        print('%d-th percentile (%d) error %f' %
-                (percentile, sample, error_rate))
-        if error_rate > acceptable_error_rate:
-            print('the error rate is not acceptable (> %f)' %
-                    acceptable_error_rate)
-            print('samples are as below')
-            print('\n'.join(['%d' % wss for wss in wss_collected]))
-            exit(1)
+    if error_rate > acceptable_error_rate:
+        print('the error rate is not acceptable (> %f)' %
+                acceptable_error_rate)
+        print('samples are as below')
+        print('\n'.join(['%d' % wss for wss in wss_collected]))
+        exit(1)
 
 if __name__ == '__main__':
     main()
-- 
2.47.3

